### PR TITLE
Ft site update errors

### PR DESCRIPTION
**Pull request by FaithDaka**
#### Summary of Changes (What does this PR do?)
This PR is a fix for the recurring bad request errors during site update.

#### Is this change ready to hit production in its current state?
- [x] Yes
- [ ] No

#### Status of maturity (all need to be checked before merging):

- [ ] I consider this code done
- [x] I've tested this locally
- [ ] This branch has been LGTM'ed by a reviewer

#### What are the relevant tickets?
- #501

### CODE REVIEW

**Review by FaithDaka — COMMENTED** [review]
The airqloud array was being appended to the update URL by an unknown command. The fix agreed upon was to delete it before form submission.

File: netmanager/src/views/components/Sites/SiteView.js [hunk 1]
```diff
@@ -53,6 +53,7 @@ const SiteForm = ({ site }) => {
   };
 
   const handleSubmit = async () => {
+    if(isEmpty(siteInfo.airqlouds)) { delete(siteInfo.airqlouds) } //empty airqloud array deleted before submit
```
Comment (Baalmart) [inline netmanager/src/views/components/Sites/SiteView.js:56]: Thanks for this. @FaithDaka , great initiative!
Could we perhaps employ some library which removes all empty fields before submission?
Just to cater for future occurrences like this one? An example library is **clean-deep**: https://www.npmjs.com/package/clean-deep
**Context:**
I am thinking about a solution which can be able to address any future occurrences of a similar problem. So that we can completely erase this problem and prevent it from ever happening again. :)

File: netmanager/src/views/components/DataDisplay/Map/MapKey.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const MapKey = ({ title }) => {
         >
           <span>{title || "map key"}</span>
           <div className={"mapkey-icon-container"} onClick={toggleIsOpen}>
-            {isOpen ? <ExpandLessIcon /> : <ExpandMoreIcon />}
+            {isOpen ? <ExpandMoreIcon /> : <ExpandLessIcon />}
```
Comment (Baalmart) [inline netmanager/src/views/components/DataDisplay/Map/MapKey.js:42]: Just wondering, is this change related to the PR purpose?

File: netmanager/src/views/components/DataDisplay/Map/MapFilter.js [hunk 3]
```diff
@@ -18,19 +18,28 @@ import "assets/css/map-filter.css";
...
-        d="M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z"
-        stroke={stroke || "white"}
-        fill={fill || "white"}
-      />
+      stroke={stroke || "white"}
+      fill={fill || "white"}/>
     </svg>
+    // <svg
+    //   xmlns="http://www.w3.org/2000/svg"
+    //   height="24"
```
Comment (Baalmart) [inline netmanager/src/views/components/DataDisplay/Map/MapFilter.js:32]: We should all avoid adding commented code to our repo. Unless it is really really necessary. Thanks.

File: netmanager/src/views/components/DataDisplay/Map/MapFilter.js [hunk 4]
```diff
@@ -18,19 +18,28 @@ import "assets/css/map-filter.css";
 
 const FilterIcon = ({ fill, stroke }) => {
   return (
```
Comment (Baalmart) [inline netmanager/src/views/components/DataDisplay/Map/MapFilter.js:20]: I am just wondering if this change is related to the PR description. Isn't this change meant/covered in the map filter PR? Just curious, I could be wrong about the PR's scope.

File: netmanager/src/assets/css/map-filter.css [hunk 5]
```diff
@@ -38,19 +38,38 @@
 }
 
 .map-filter-searchButton {
-  width: 40px;
-  height: 36px;
+  width: 48px;
```
Comment (Baalmart) [inline netmanager/src/assets/css/map-filter.css:41]: Is this change part of this PR's scope? Could we be mixing up work of different branches/PRs? Just curious....awaiting your feedback.

File: netmanager/src/views/components/Sites/SiteView.js (hunk 1 above)
Comment (FaithDaka) [reply netmanager/src/views/components/Sites/SiteView.js:56]: Yes, I'd thought of that as well. Being able to drop empty fields dynamically would be great, though I wanted to first find where those fields come from in the first place. Because as per the code, only data from changed input fields is posted to the server so how do certain empty fields end up appended to the route as well?
Let me check out the library. Thank you!

File: netmanager/src/assets/css/map-filter.css (hunk 5 above)
Comment (FaithDaka) [reply netmanager/src/assets/css/map-filter.css:41]: I checked out from a branch I thought had been merged, instead of checking out from staging.
I think let me delete this branch and re-push?
